src/MPI.jl

```diff
@@ -103,6 +103,10 @@ function __init__()
     end
 
     @static if Sys.isunix()
+        # Cray MPICH apparently requires you to manually load a separate GPU Transport Layer (GTL) if you want GPU-aware MPI.
+        if MPIPreferences.binary == "system" && MPIPreferences.abi == "MPICH" && get(ENV, "MPICH_GPU_SUPPORT_ENABLED", "0") == "1"
+            Libdl.dlopen_e(dirname(libmpi), "libmpi_gtl_cuda"), Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
```

Comment:
This assumes that the library is in the same directory as `libmpi`: is this the case @JBlaschke?

Also, is there any way to determine whether we should load `libmpi_gtl_cuda` or `libmpi_gtl_hsa` (`dlopen_e` should fail silently, so we could just try opening both?)